tao_compiler/mlir/disc/transforms/codegen_utils.cc

```diff
@@ -44,17 +47,27 @@ int getRowReductionScheduleHint(Operation* op) {
 
 int getVectorizeOrTileHint(Operation* op) {
   assert(op);
-  lmhlo::FusionOp fusion = op->getParentOfType<lmhlo::FusionOp>();
-  if (!fusion) return 1;
+  lmhlo::FusionOp fusion = dyn_cast<lmhlo::FusionOp>(op);
+  if (!fusion) {
```

Comment:
when would `fusion` be null?

Reply:
If `op` is not a `lmhlo::FusionOp`, the cast returns null.